Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM rust:1.70-slim-bullseye
```

Comment:
This version seems to have a high vulnerability. Probably this is very recent. 

https://hub.docker.com/layers/library/rust/1.70-slim-bullseye/images/sha256-53f49ce140a3d19c775da59c3d9b124c1830425ef5568ca45d1f29e76cb15f96?context=repo&tab=vulnerabilities

We should check if there is an updated version.

Reply:
Ahhh, I had never paid attention to the image vulnerability before. Thank you for pointing it out. It seems `bookworm` is the better one, I'll replace the base image with it.

Reply:
We hit the mine again =D 

1.70-slim-bookworm has the same vulnerability too. But it seems for different OS/ARCH, it's different.

https://hub.docker.com/layers/library/rust/1.70-slim-bookworm/images/sha256-162271b91daf42c769cb537ff6ec14bfd35a53e8392c8a30686c9b2368d25acb?context=repo&tab=vulnerabilities

This is what I see when I open it. Probably it defaults for my Mac M1 differently: 
![image](https://github.com/burn-rs/burn/assets/939125/2044b89a-bff2-4682-b59f-45385e092219)

Reply:
Emmm...It seems almost all Debian release have been suffering the poor vulnerability problem.

Since we also need GPU support, maybe we should turn to PyTorch image as the base.

The difference is to solve rust toolchain (use `1.70-slim-xxx`) or CUDA/CuDNN dependencies (PyTorch tags).

I think finally an elegant way will come with more experiments:thinking: